collect_app/src/main/java/org/odk/collect/android/tasks/InstanceGoogleSheetsUploader.java

```diff
@@ -71,78 +80,103 @@
     private static final String oauth_fail = "OAUTH Error: ";
     private static final String UPLOADED_MEDIA_URL = "https://drive.google.com/open?id=";
 
-    // needed in case of rate limiting
-    private static final int GOOGLE_SLEEP_TIME = 1000;
-
     private final SheetsHelper sheetsHelper;
     private final DriveHelper driveHelper;
-
     private final GoogleAccountsManager accountsManager;
-    private Integer sheetId;
-    private String sheetName;
     private Outcome outcome;
-    private boolean hasWritePermissionToSheet = false;
+
+    private boolean hasWritePermissionToSheet;
```

Comment:
Is this field actually useful? It's only ever set to true so I think that means even if you're uploading a bunch of different forms to different sheets, as soon as you hit the first one you have write access to the field is true. I think it can be removed.